src/Passes/Source/Rules/OperandPrototype.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Rules/OperandPrototype.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    IOperandPrototype::~IOperandPrototype() = default;
+    bool IOperandPrototype::matchChildren(Value* value, Captures& captures) const
+    {
+        auto user = llvm::dyn_cast<llvm::User>(value);
+        if (!children_.empty())
+        {
+            if (user == nullptr)
+            {
+                return false;
+            }
+
+            if (user->getNumOperands() != children_.size())
+            {
+                return false;
+            }
+
+            uint64_t i = 0;
+            while (i < children_.size())
+            {
+                auto v = user->getOperand(static_cast<uint32_t>(i));
+                if (!children_[i]->match(v, captures))
+                {
+                    return false;
+                }
+                ++i;
+            }
+
+            return true;
+        }
+
+        // TODO(QAT-private-issue-33): value may be other type than llvm::User. Check other relevant types
+        // and deal with it.
+
+        return true;
+    }
+
+    void IOperandPrototype::addChild(Child const& child)
+    {
+        children_.push_back(child);
+    }
+
+    void IOperandPrototype::enableCapture(std::string capture_name)
+    {
+        capture_name_ = std::move(capture_name);
+    }
+
```

Comment:
The name `enableCapture()` makes me expect that some Boolean state is to be changed.
But the implementation makes me expect that the name should be something like `setCaptureName()` (rather than `enableCapture()`).
Consider renaming.

Reply:
I've renamed to `captureAs` as I think this is more accurate since the default behaviour is that variables are not captured.